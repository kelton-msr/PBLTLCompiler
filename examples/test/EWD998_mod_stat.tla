-------------------------- MODULE EWD998_mod_stat ---------------------------

EXTENDS Integers, Functions, CSV, EWD998, Sequences

VARIABLES hasEmittedCSV, c
\*Property == 
\*    [](((c["temp"]["counter"] <= 8) \/ c["temp"]["condition"]))

Foo == INSTANCE EWD998

fvars == <<vars,c, hasEmittedCSV>>

cond == c["temp"]["counter"] > 3


FNext == 
    /\ Foo!Next
    /\ c' = [temp|-> [condition|-> (c["temp"]["condition"] \/ Foo!terminationDetected), counter|-> (IF ~(c["temp"]["hasAntecedent"]) THEN 0 ELSE (IF (c["temp"]["condition"] /\ Foo!Termination) THEN 1 ELSE (c["temp"]["counter"] + 1))), hasAntecedent|-> (c["temp"]["hasAntecedent"] \/ Foo!Termination)]]
    /\ hasEmittedCSV' = ~(((c'["temp"]["counter"] <= 8) \/ c'["temp"]["condition"]))
    /\ (IF (hasEmittedCSV' /\ ~(hasEmittedCSV)) THEN CSVWrite("%1$s", <<TRUE>>, "statPropsRes.csv") ELSE TRUE)

FInit ==
    /\ Foo!Init 
    /\ hasEmittedCSV = FALSE
    /\ c = [temp|-> [counter|-> 0, condition|-> Foo!terminationDetected, hasAntecedent|-> Foo!Termination]]

FSpec == FInit /\ [][FNext]_fvars /\ WF_fvars(FNext)

======
