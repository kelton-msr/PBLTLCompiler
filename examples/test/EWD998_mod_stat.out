@!@!@STARTMSG 2262:0 @!@!@
TLC2 Version 2.18 of Day Month 20?? (rev: a57a3a7)
@!@!@ENDMSG 2262 @!@!@
@!@!@STARTMSG 2187:0 @!@!@
Running breadth-first search Model-Checking with fp 65 and seed -8229090852850035065 with 1 worker on 16 cores with 3422MB heap and 64MB offheap memory [pid: 54257] (Linux 6.1.38-1-MANJARO amd64, N/A 17.0.7 x86_64, MSBDiskFPSet, DiskStateQueue).
@!@!@ENDMSG 2187 @!@!@
@!@!@STARTMSG 2220:0 @!@!@
Starting SANY...
@!@!@ENDMSG 2220 @!@!@
Parsing file /home/kelton/work/PBLTLCompiler/examples/test/EWD998_mod_stat.tla
Parsing file /tmp/tlc-6880467171064635941/Integers.tla (jar:file:/home/kelton/.vscode/extensions/alygin.vscode-tlaplus-nightly-2023.8.906/tools/tla2tools.jar!/tla2sany/StandardModules/Integers.tla)
Parsing file /tmp/tlc-6880467171064635941/Functions.tla (jar:file:/home/kelton/.vscode/extensions/alygin.vscode-tlaplus-nightly-2023.8.906/tools/CommunityModules-deps.jar!/Functions.tla)
Parsing file /tmp/tlc-6880467171064635941/CSV.tla (jar:file:/home/kelton/.vscode/extensions/alygin.vscode-tlaplus-nightly-2023.8.906/tools/CommunityModules-deps.jar!/CSV.tla)
Parsing file /home/kelton/work/PBLTLCompiler/examples/test/EWD998.tla
Parsing file /tmp/tlc-6880467171064635941/Sequences.tla (jar:file:/home/kelton/.vscode/extensions/alygin.vscode-tlaplus-nightly-2023.8.906/tools/tla2tools.jar!/tla2sany/StandardModules/Sequences.tla)
Parsing file /tmp/tlc-6880467171064635941/Naturals.tla (jar:file:/home/kelton/.vscode/extensions/alygin.vscode-tlaplus-nightly-2023.8.906/tools/tla2tools.jar!/tla2sany/StandardModules/Naturals.tla)
Parsing file /tmp/tlc-6880467171064635941/FiniteSets.tla (jar:file:/home/kelton/.vscode/extensions/alygin.vscode-tlaplus-nightly-2023.8.906/tools/tla2tools.jar!/tla2sany/StandardModules/FiniteSets.tla)
Parsing file /tmp/tlc-6880467171064635941/SequencesExt.tla (jar:file:/home/kelton/.vscode/extensions/alygin.vscode-tlaplus-nightly-2023.8.906/tools/CommunityModules-deps.jar!/SequencesExt.tla)
Parsing file /tmp/tlc-6880467171064635941/Randomization.tla (jar:file:/home/kelton/.vscode/extensions/alygin.vscode-tlaplus-nightly-2023.8.906/tools/tla2tools.jar!/tla2sany/StandardModules/Randomization.tla)
Parsing file /tmp/tlc-6880467171064635941/Folds.tla (jar:file:/home/kelton/.vscode/extensions/alygin.vscode-tlaplus-nightly-2023.8.906/tools/CommunityModules-deps.jar!/Folds.tla)
Parsing file /tmp/tlc-6880467171064635941/TLC.tla (jar:file:/home/kelton/.vscode/extensions/alygin.vscode-tlaplus-nightly-2023.8.906/tools/tla2tools.jar!/tla2sany/StandardModules/TLC.tla)
Parsing file /home/kelton/work/PBLTLCompiler/examples/test/AsyncTerminationDetection.tla
Parsing file /tmp/tlc-6880467171064635941/FiniteSetsExt.tla (jar:file:/home/kelton/.vscode/extensions/alygin.vscode-tlaplus-nightly-2023.8.906/tools/CommunityModules-deps.jar!/FiniteSetsExt.tla)
Semantic processing of module Naturals
Semantic processing of module Integers
Semantic processing of module Folds
Semantic processing of module Functions
Semantic processing of module Sequences
Semantic processing of module FiniteSets
Semantic processing of module TLC
Semantic processing of module CSV
Semantic processing of module FiniteSetsExt
Semantic processing of module SequencesExt
Semantic processing of module Randomization
Semantic processing of module AsyncTerminationDetection
Semantic processing of module EWD998
Semantic processing of module EWD998_mod_stat
@!@!@STARTMSG 2219:0 @!@!@
SANY finished.
@!@!@ENDMSG 2219 @!@!@
@!@!@STARTMSG 2185:0 @!@!@
Starting... (2023-08-09 14:06:13)
@!@!@ENDMSG 2185 @!@!@
@!@!@STARTMSG 2189:0 @!@!@
Computing initial states...
@!@!@ENDMSG 2189 @!@!@
@!@!@STARTMSG 2269:0 @!@!@
Computed 2 initial states...
@!@!@ENDMSG 2269 @!@!@
@!@!@STARTMSG 2269:0 @!@!@
Computed 4 initial states...
@!@!@ENDMSG 2269 @!@!@
@!@!@STARTMSG 2269:0 @!@!@
Computed 8 initial states...
@!@!@ENDMSG 2269 @!@!@
@!@!@STARTMSG 2269:0 @!@!@
Computed 16 initial states...
@!@!@ENDMSG 2269 @!@!@
@!@!@STARTMSG 2269:0 @!@!@
Computed 32 initial states...
@!@!@ENDMSG 2269 @!@!@
@!@!@STARTMSG 2269:0 @!@!@
Computed 64 initial states...
@!@!@ENDMSG 2269 @!@!@
@!@!@STARTMSG 2269:0 @!@!@
Computed 128 initial states...
@!@!@ENDMSG 2269 @!@!@
@!@!@STARTMSG 2190:0 @!@!@
Finished computing initial states: 256 distinct states generated at 2023-08-09 14:06:14.
@!@!@ENDMSG 2190 @!@!@
@!@!@STARTMSG 2200:0 @!@!@
Progress(4) at 2023-08-09 14:06:17: 90,468 states generated (90,468 s/min), 32,633 distinct states found (32,633 ds/min), 24,766 states left on queue.
@!@!@ENDMSG 2200 @!@!@
@!@!@STARTMSG 2200:0 @!@!@
Progress(7) at 2023-08-09 14:07:17: 2,269,231 states generated (2,178,763 s/min), 443,622 distinct states found (410,989 ds/min), 277,570 states left on queue.
@!@!@ENDMSG 2200 @!@!@
